doc/release/upcoming_changes/21627.new_feature.rst

```diff
@@ -0,0 +1,16 @@
+``casting`` and ``dtype`` keyword arguments for `numpy.stack`
+-------------------------------------------------------------
+The ``casting`` and ``dtype`` keyword arguments is now available for `numpy.stack`. 
+To use it, write ``np.stack(..., dtype=None, casting='same_kind')``.
```

Comment:
```suggestion
To use them, write ``np.stack(..., dtype=None, casting='same_kind')``.
```

Reply:
Thanks @melissawm, we already pushed the commits with those changes!